manual/src/refman/extensions/arrayliterals.etex

```diff
@@ -0,0 +1,74 @@
+(Introduced in OCaml 5.4)
+
+Since OCaml 5.4, array literal syntax @"[|" e1 ';' e2 ';' ... ';' eN "|]"@ can
+be used to denote values of type "'a array" as well as "floatarray", both in
+expression and pattern positions. This syntax is also used to display "floatarray" values in the toplevel.
+
+The compiler will use contextual information to infer the type of the array
+literal. However, in the absence of any specific contextual information, it will
+be assumed that the array literal is of type "'a array":
+
+\begin{caml_example}{verbatim}
+let x = [|42.|];;
+\end{caml_example}
+
+The compiler will exploit the knowledge it gleans from the context of the array
+literal to decide which type to assign to it. In the following examples,
+the array literal is assigned type "floatarray":
+
+\begin{caml_example}{verbatim}
+let _ : floatarray = [|42.|];;
+let _ = ([|42.|] : floatarray);;
+let _ = Float.Array.length [|42.|];;
+\end{caml_example}
+
+The same disambiguation mechanism is used for array literals appearing in
+patterns:
+
+\begin{caml_example}{verbatim}
+let f : floatarray -> _ = function
+  | [| 42. |] -> "It's a floatarray containing one element"
+  | _ -> "Also a floatarray"
+;;
+\end{caml_example}
+
+A type annotation is not strictly necessary; the compiler will use any type
+information that it learns from the context of the array literal to disambiguate
+it. In the next example, the use of "Float.Array.get" tells the compiler that
+"a" must be of type "floatarray" and the array literal pattern is interpreted
+accordingly:
```

Comment:
I am not sure we need this section, and the term `any type information that it learns from the context` is too imprecise and is contradicted by your next example.

Reply:
I removed this paragraph. Thanks.